File: prototype/rails/src/test/domain/shop/shop_name_test.rb
require "test_helper"

class ShopNameTest < ActiveSupport::TestCase
  ShopName = Domain::Shop::ShopName

  test "a named shop reads by its name" do
    assert_equal "Blue Kiln Studio", ShopName.of(shop_name: "Blue Kiln Studio", email: "ada@example.test")
  end

  test "an unnamed shop reads by the address" do
    assert_equal "ada", ShopName.of(shop_name: nil, email: "ada@example.test")
  end

  test "a blank name reads by the address" do
    assert_equal "ada", ShopName.of(shop_name: "   ", email: "ada@example.test")
  end
end

File: prototype/node/src/app/core/shop/shop-name.test.ts
import { test } from 'node:test'
import assert from 'node:assert/strict'
import { shopName } from './shop-name.ts'

test('a named shop reads by its name', () => {
  assert.equal(shopName({ shopName: 'Blue Kiln Studio', email: 'ada@example.test' }), 'Blue Kiln Studio')
})

test('an unnamed shop reads by the address', () => {
  assert.equal(shopName({ shopName: null, email: 'ada@example.test' }), 'ada')
})

test('a blank name reads by the address', () => {
  assert.equal(shopName({ shopName: '   ', email: 'ada@example.test' }), 'ada')
})
